core/Symbols.cc

```diff
@@ -818,18 +818,15 @@ bool Symbol::isSingletonClass(const GlobalState &gs) const {
     return isSingleton;
 }
 
-SymbolRef Symbol::singletonClass(GlobalState &gs) {
-    auto singleton = lookupSingletonClass(gs);
-    if (singleton.exists()) {
-        return singleton;
-    }
+SymbolRef Symbol::defineSingletonClass(GlobalState &gs) {
```

Comment:
Note: This is a private method. GlobalState can call it because it's a friend of GlobalState.

I debated calling it "unsafeDefineSingletonClass".